python/pip_install/tools/dependency_resolver/dependency_resolver.py

```diff
@@ -99,27 +103,7 @@ def _fix_up_requirements_in_path(absolute_prefix, output_file):
         requirements_out = os.path.join(
             os.environ["TEST_TMPDIR"], os.path.basename(requirements_txt) + ".out"
         )
-        copyfile(requirements_txt, requirements_out)
-
-    elif "BUILD_WORKSPACE_DIRECTORY" in os.environ:
-        # This value, populated when running under `bazel run`, is a path to the
-        # "root of the workspace where the build was run."
-        # This matches up with the values passed in via the macro using the 'rootpath' Make variable,
-        # which for source files provides a path "relative to your workspace root."
-        #
-        # Changing to the WORKSPACE root avoids 'file not found' errors when the `.update` target is run
-        # from different directories within the WORKSPACE.
-        os.chdir(os.environ["BUILD_WORKSPACE_DIRECTORY"])
-    else:
-        err_msg = (
-            "Expected to find BUILD_WORKSPACE_DIRECTORY (running under `bazel run`) or "
-            "TEST_TMPDIR (running under `bazel test`) in environment."
-        )
-        print(
-            err_msg,
-            file=sys.stderr,
-        )
-        sys.exit(1)
+        shutil.copy(requirements_txt, requirements_out)
```

Comment:
comment about how this actually copies back over the source file because we assume it's a symlink, and that's why we don't have to lookup the BUILD_WORKSPACE_DIRECTORY after all?

Also what happens if the target file didn't exist? I guess we fail in the analysis phase and don't even run this.